build/tools/Automation/Services/VersionService.cs

```diff
@@ -21,9 +21,10 @@ public class VersionService : IVersionService
         /// <returns>True if the version is within the specified range and not in the list of blocked versions. False otherwise.</returns>
         public bool IsVersionWithinRange(string version, string minVersion = null, string maxVersion = null, List<string> blockedVersions = null)
         {
-            // Try to parse the version string into a SemanticVersion object and ignores pre-releases
+            // Try to parse the version string into a SemanticVersion object and ignores pre-releases that are not dotnet 8
+            // Block pre release once stabled. Workitem: https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1891562
```

Comment:
Could you please share more info about what prevents us from having a logic that would work for both the pre-releases and the GA'd versions?

Reply:
For this one, this is more to just include prereleases for dotnet 8. But this check doesn't prevent us from getting GA'd versions. And because we need to make sure things are compatible when new major version releases, it would be better to limit the prereleases to just the versions we want to ingest rather than having it taking all prereleases imo.

Reply:
OK sorry I want to be sure I understand: what prereleases do we expect outside of the .NET 8 ones? Do they provided other packages labeled as prereleases that we don't care about?

Reply:
Oh i see. There's no other prereleases outside of .NET 8 for now. But say if .NET 9 is out, this would prevent it accidentally takes in the prereleases of the hypothetical .NET 9. Just a preventive measure here.

Reply:
Thanks! At the same time having .NET 9 pre-releases would sound good pretty good to me as well. :D Let's sync separately